repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -97,6 +99,8 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
   private var keyEntry: KeyStore.PrivateKeyEntry = _
   private var keyInfo: KeyInfo = _
   private var legacyIssuers: List[URI] = List.empty
+  private val documentBuilderFactory = DocumentBuilderFactory.newInstance()
+  documentBuilderFactory.setNamespaceAware(true)
```

Comment:
Please consider using XML Parser pool instead of doing this from scratch. 

For a couple of reasons:

1. It's more efficient to reuse a builder factory.
2. You need to take special security issues into mind when parsing XML.  There are attacks that are possible in the normal configuration.
3.  There are potential class loader issues in different deployment if you simply do ```newInstance()``` on a document builder factory  also some versions of ```newInstance()``` scan jars which makes them really slow.


You can find parsing pool here.

https://github.com/rackerlabs/api-checker/blob/master/util/src/main/scala/com/rackspace/com/papi/compenents/checker/util/XMLParserPool.scala#L25

It handles all of the issues above and has been battle tested.

Use ```borrowParser``` to get ```DocumentBuilder``` and ```returnParser``` when you're done -- usually do this in a ```try {} finally {} ``` 

Here's a simple test that  shows the pattern:

https://github.com/rackerlabs/api-checker/blob/master/util/src/test/scala/com/rackspace/com/papi/components/checker/util/XMLParserPoolSuite.scala#L27